google/cloud/forseti/services/inventory/base/cai_gcp_client.py

```diff
@@ -76,39 +76,23 @@ def _fixup_resource_keys(resource, key_map, only_fixup_lists=False):
 class CaiApiClientImpl(gcp.ApiClientImpl):
     """The gcp api client Implementation"""
 
-    def __init__(self, config, engine, parallel, session):
+    def __init__(self, config, engine, tmpfile):
         """Initialize.
 
         Args:
             config (dict): GCP API client configuration.
             engine (object): Database engine to operate on.
-            parallel (bool): If true, use the parallel crawler implementation.
-            session (object): Database session.
+            tmpfile (str): The temporary file storing the cai sqlite database.
         """
         super(CaiApiClientImpl, self).__init__(config)
         self.dao = CaiDataAccess()
         self.engine = engine
-        self.parallel = parallel
-        self.cai_session = session
-        self._local = LOCAL_THREAD
+        self.tmpfile = tmpfile
 
-    @property
-    def session(self):
-        """Return a thread local CAI read only session object.
-
-        Returns:
-            object: A thread local Session.
-        """
-        if not self.parallel:
-            # SQLite doesn't support per thread sessions cleanly, so use global
-            # session.
-            return self.cai_session
-
-        if hasattr(self._local, 'cai_session'):
-            return self._local.cai_session
-
-        self._local.cai_session = db.create_readonly_session(engine=self.engine)
-        return self._local.cai_session
+    def __del__(self):
```

Comment:
Can't seem to find where the file is being cleaned up, where do we plan to call this method?

Reply:
This is called as soon as the class goes out of scope in crawler.run_crawler. It's called by the garbage collector.